storageClient/api.go

```diff
@@ -50,6 +50,19 @@ type DescribeGroupRequest struct {
 	GroupID uuid.UUID `json:"group_id"`
 }
 
+// ListGroupsRequest wraps values used to Look up Groups a client is associated with
+type ListGroupsRequest struct {
+	ClientID  uuid.UUID `json:"client_id"`
+	NextToken int64     `json:"next_token"`
+	Max       int       `json:"max"`
+}
+
+// ListGroupsResponse wraps the list of groups a client is associated with
```

Comment:
`wraps a collection of groups`, associated with a client is maybe the only filter we have for today but in the future we could have other